homeassistant/core.py

```diff
@@ -132,6 +132,7 @@
 
 # How long we wait for the result of a service call
 SERVICE_CALL_LIMIT = 10  # seconds
+ServiceResult = bool | dict[str, Any] | None
```

Comment:
```suggestion
ServiceResult = bool | JsonObjectType | None
```

Reply:
I think we should avoid `bool`? here completely? Having multiple possible types returned, makes handling in template unneeded difficult (as the user has to evaluate the type before use).

If there is an error, it should raise already. Otherwise, it would succeed.

Similar story with `None`, if one sets a variable to get a return value, I think a JsonObjectType should always be returned. If the result is `None`, I think we could raise it, as that was not expected.

Reply:
I agree the right API is to not have a return boolean.  As I said in the PR description: "The existing return value of True has been preserved when not opting into return values (though we can consider removing it entirely as an alternative given lack of an exception is now success).".   I interpret your comment to be less conservative and in favor of a breaking change to get us faster to the right API.

My impression is the implications are:
- Within core, I believe this only has implications for tests that assert True. No actual code dependencies outside of tests check the return value at the moment
- Outside of core, is it worth a blog post?

If so, then I would propose I solve this with a "prefactor" PR then come back to this.

Do I understand that right?

Reply:
It makes sense to be. It isn't really used, we can use a small little dev blog post to notify custom integration developers (although I would be surprised if it was used intentionally).

../Frenck